src/z3c/authviewlet/session.py

```diff
@@ -27,10 +29,7 @@ def __init__(self, *args):
             self.request.principal)
 
         camefrom = self.request.get('camefrom')
-        if isinstance(camefrom, list):
-            # this can happen on python2.6, as it changed the
-            # behaviour of cgi.FieldStorage a bit.
-            camefrom = camefrom[0]
+        assert isinstance(camefrom, basestring), type(camefrom)
```

Comment:
Can't this happen on 2.7+ too, if the user edits the location bar and adds a second `&camefrom=somewhere-else-too`?

I don't think users should be able to trigger internal server errors by monkeying around with URLs.

Reply:
You are right, I was misguided by the comment. I am going to switch back to the original code and add a test.